src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -1,67 +1,211 @@
-      .setServiceAccountId(dataprocConfig.serviceAccount)
+      .setServiceAccountId(dc.serviceAccount)
       .setServiceAccountScopes(cloudPlatformScopes.asJava)
-      .setServiceAccountPrivateKeyFromPemFile(new java.io.File(dataprocConfig.serviceAccountPemPath))
+      .setServiceAccountPrivateKeyFromPemFile(serviceAccountPemFile)
+      .build()
+  }
+
+  def getBucketServiceAccountCredential: Credential = {
+    new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(dc.serviceAccount)
+      .setServiceAccountScopes(storageScopes) // grant bucket-creation powers
+      .setServiceAccountPrivateKeyFromPemFile(serviceAccountPemFile)
       .build()
   }
 
+  private def getVmServiceAccountCredential: Credential = {
+    new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(dc.serviceAccount)
+      .setServiceAccountScopes(vmScopes.asJava)
+      .setServiceAccountPrivateKeyFromPemFile(serviceAccountPemFile)
+      .build()
+  }
 
   def createCluster(googleProject: String, clusterName: String, clusterRequest: ClusterRequest)(implicit executionContext: ExecutionContext): Future[ClusterResponse] = {
-    val op = build(googleProject, clusterName, clusterRequest)
-    op.map{op =>
-      val metadata = op.getMetadata
-      ClusterResponse(clusterName, googleProject, metadata.get("clusterUuid").toString, metadata.get("status").toString, metadata.get("description").toString, op.getName)}
+    val bucketName = s"${clusterName}-${UUID.randomUUID.toString}"
+
+    val bucketResponse = initializeBucket(googleProject, clusterName, bucketName, clusterRequest.serviceAccount)
+
+    bucketResponse.flatMap[ClusterResponse]{ Bucket =>
+      val op = build(googleProject, clusterName, clusterRequest, bucketName)
+      op.map { op =>
+        val metadata = op.getMetadata
+        ClusterResponse(clusterName, googleProject, metadata.get("clusterUuid").toString, metadata.get("status").toString, metadata.get("description").toString, op.getName)
+      }
+    }
+  }
+
+  private def createBucketUri(bucketName: String, fileName: String): String = {
+    s"gs://${bucketName}/${fileName}"
+  }
+
+  private def initializeBucket(googleProject: String, clusterName: String, bucketName:String, serviceAccount: String): Future[Bucket] = {
+    import org.broadinstitute.dsde.workbench.leonardo.model.LeonardoJsonSupport._
+    import spray.json._
+    Future {
+      val bucket = new Bucket().setName(bucketName)
+      val bucketInserter = getStorage(getBucketServiceAccountCredential).buckets().insert(googleProject, bucket)
+      try {
+       executeGoogleRequest(bucketInserter) //returns a Bucket
+      } catch {
+        case e: GoogleJsonResponseException => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"Google Request Failed: ${e.getDetails.getMessage}"))
+      } finally {
```

Comment:
I'm not sure this stuff should go in `finally`. If the bucket creation fails, how can we write to it?